python/social_bot/tasks.py

```diff
@@ -93,6 +87,36 @@ def task_specific_observation(self, agent):
         """
         return np.array([])
 
+    def set_agent(self, agent):
+        """
+        The agent can be override by this function. 
+        This might be useful when multi agents share the same task or embodied teacher.
+        Args:
+            agent (Agent): the agent
+        """
+        self._agent = agent
+
+    def _get_states_of_model_list(self, model_list, including_rotation=False):
+        """
+        Get the poses and velocities for the models
+        Args:
+            model_list (list): a list of model names
+            including_rotation (bool): if Ture, the rotation of objects (in roll pitch yaw) will be included.
+        Returns:
+            np.array, the poses and velocities of the models
+        """
+        model_poss = []
+        model_vels = []
+        for model_id in range(len(model_list)):
+            model = self._world.get_model(model_list[model_id])
+            model_poss.append(model.get_pose()[0])
+            if including_rotation:
+                model_poss.append(model.get_pose()[1])
+            model_vels.append(model.get_velocities()[0])
```

Comment:
if including_rotation:

Reply:
Rotating velocity is not useful. 
But sometimes linear velocity is also unwanted. Let's add another option "including_velocity"